upgrade/scripts/upgrade/ncn-upgrade-master-nodes.sh

```diff
@@ -205,14 +205,31 @@ if [[ $state_recorded == "0" ]]; then
   } >> ${LOG_FILE} 2>&1
   record_state "${state_name}" ${target_ncn}
 else
-  echo "====> ${state_name} has been completed"
+  echo "INFO ====> ${state_name} has been completed"
 fi
 
 cat << EOF
 NOTE:
     If below test failed, try to fix it based on test output. Then run current script again
 EOF
-ssh $target_ncn -t 'GOSS_BASE=/opt/cray/tests/install/ncn goss -g /opt/cray/tests/install/ncn/suites/ncn-upgrade-tests-master.yaml --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate'
+state_name="RUN_GOSS_TESTS"
+state_recorded=$(is_state_recorded "${state_name}" ${target_ncn})
+if [[ $state_recorded == "0" ]]; then
+  echo "====> ${state_name} ..."
+  {
+    goss_output=$(ssh $target_ncn -t "GOSS_BASE=/opt/cray/tests/install/ncn goss -g /opt/cray/tests/install/ncn/suites/ncn-upgrade-tests-master.yaml --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate")
+    echo ${goss_output} | while read -r line; do
+      if [[ -n $(echo $line | grep 'Title\|desc') ]]; then
+        echo "ERROR failed goss test: $line"
+      else
+        echo "$line"
+      fi
+    done
+  } | tee -a ${LOG_FILE} 2>&1
+  record_state "${state_name}" ${target_ncn}
+else
+  echo "INFO ====> ${state_name} has been completed"
+fi
```

Comment:
This should be replaced with the following.
```suggestion
state_name="RUN_GOSS_TESTS"
state_recorded=$(is_state_recorded "${state_name}" ${target_ncn})
if [[ $state_recorded == "0" ]]; then
  echo "====> ${state_name} ..."
  {
    set +e
    return_code=0
    while read line; do
      if [[ -n $(echo $line | grep 'Title\|desc') ]]; then
        echo "ERROR failed goss test: $line"
	return_code=1
      else
        echo -e "$line"
      fi
done < <(ssh $target_ncn -t "GOSS_BASE=/opt/cray/tests/install/ncn goss -g /opt/cray/tests/install/ncn/suites/ncn-upgrade-tests-master.yaml --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate")
    if [[ $return_code -eq 1 ]]; then
      exit $return_code
    fi
  } | tee -a ${LOG_FILE} 2>&1
  record_state "${state_name}" ${target_ncn}
else
  echo "INFO ====> ${state_name} has been completed"
fi
```
Changes are 
- set +e: so the segment doesn't exit if goss tests return an error
- read lines directly from goss test output. The goss_output variable was not receiving all of the output from the goss tests. 
- add a return_code so that the segment will exit if goss tests don't pass.